src/engine/strat_engine/blockdev.rs

```diff
@@ -55,10 +56,67 @@ pub fn resolve_devices(paths: &[&Path]) -> io::Result<BTreeSet<Device>> {
     Ok(devices)
 }
 
+#[derive(Debug)]
+pub struct BlockDevs {
+    pub devs: BTreeMap<DevUuid, BlockDev>,
+}
+
+impl BlockDevs {
+    pub fn new(devs: BTreeMap<DevUuid, BlockDev>) -> BlockDevs {
+        BlockDevs { devs: devs }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.devs.is_empty()
+    }
+
+    /// Read the latest json across all blockdevs
+    pub fn read_metadata(&self) -> Option<Vec<u8>> {
+
+        let mut bds: Vec<&BlockDev> = self.devs
+            .iter()
+            .map(|(_, bd)| bd)
+            .filter(|bd| bd.bda.last_updated().is_some())
+            .collect();
+
+        bds.sort_by_key(|k| k.bda.last_updated().unwrap());
+
+        for bd in bds.iter().rev() {
+            let mut f = match OpenOptions::new()
+                .read(true)
+                .open(&bd.devnode) {
+                Ok(f) => f,
+                Err(_) => continue,
+            };
+            match bd.bda.load_state(&mut f) {
+                Ok(Some(json)) => return Some(json),
+                _ => continue,
```

Comment:
What would we do differently? We could log something if `Err` but the real problem arises if there are *no* good copies to be found. I could see us getting fancy and writing more copies if we didn't succeed getting enough, but that is an enhancement. I'll add a TODO.

I also am going to change this so we only look at metadata with the latest timestamp. Anything previous is out of date and pretty much useless...